### PR TITLE
feature 개수 오류

### LINKED ISSUE #7: data feature 개수 오류

### PR DESCRIPTION
**Pull request by byb-8**
## 변경 사항
 dataloader와 data_function에서 train_data와 test_data의 feature 개수가 맞지않아서 고쳤고, 오를 지 내릴지 0과1로 구분할 수 있도록 label을 수정하였습니다.
model.py에서 원래는 종가를 예측할려고 했지만 오를 지 내릴 지 예측하는 AI로 변경되어 그에 맞는 모델로 변경하였습니다.

### CODE REVIEW

**Comment by kdyeon0309** [comment]
해당 모델을 돌려 보셨나요?

File: stock_price_forecast/dataloader.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import torch
+import numpy as np
+import pandas as pd
+import time
+from torch.utils.data import TensorDataset, DataLoader
+from data_function import *
+
+def get_dataset():
+    markets = ["KRW-BTC", "KRW-ETH", "KRW-XRP", "KRW-SOL", "KRW-ADA"]
```
Comment (kdyeon0309) [inline stock_price_forecast/dataloader.py:9]: 현재 5가지의 종목을 선정하였는데 일단 임의로 이렇게 정하신건가요 아니면 더 수정될 예정인가요

File: stock_price_forecast/data_function.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import requests
+import pandas as pd
+import numpy as np
+
+def data_load(market, count=120):
```
Comment (kdyeon0309) [inline stock_price_forecast/data_function.py:5]: 왜 120일로 그냥 고정해 놨놔요

File: stock_price_forecast/main.py [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
+    hidden_size = 32
     num_layers = 1
     num_classes = 1
-    model=GRU(num_classes, input_size, hidden_size, num_layers, x_train.shape[1])
-    criterion = nn.MSELoss()
+    model=GRU(num_classes, input_size, hidden_size, num_layers, seq_length=3)
+    criterion = nn.BCELoss()
     optimizer = torch.optim.Adam(model.parameters(), lr=0.0001)
-    train(model, x_train, y_train, criterion, optimizer, 10000)
+    train(model, train_loader, val_loader, criterion,optimizer, 1000)
```
Comment (kdyeon0309) [inline stock_price_forecast/main.py:14]: 해당 하이퍼 파라미터들의 선정 이유들과 결과가 어땠는지 알 수 있을까요

File: stock_price_forecast/trainer.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+def test(model, x_test, y_test, criterion):
+    model.eval()
+    with torch.no_grad():
+        outputs = model(x_test)
+        predicted = (outputs >= 0.5).float()
+        y_test=y_test.squeeze(1)
+        predicted=predicted.squeeze(1)
+        correct = (predicted == y_test).sum().item()
+        accuracy = correct / y_test.size(0)
+        print(f"Accuracy: {accuracy*100:.2f}%")
```
Comment (kdyeon0309) [inline stock_price_forecast/trainer.py:56]: 테스트 후 정확도가 어떻게 나왔는지 공유 부탁 드립니다

File: stock_price_forecast/dataloader.py (hunk 1 above)
Comment (byb-8) [reply stock_price_forecast/dataloader.py:9]: 대표적인 코인들로 종목을 선정했고, 앞으로 데이터를 더 추가할 생각입니다.

File: stock_price_forecast/data_function.py (hunk 2 above)
Comment (byb-8) [reply stock_price_forecast/data_function.py:5]: 한 종목당 4개월 분량을 가져올려고 고정해뒀습니다.

File: stock_price_forecast/main.py (hunk 3 above)
Comment (byb-8) [reply stock_price_forecast/main.py:14]: 학습률은 가장 validation loss가 낮은 0.0001을 선택했고, epoch 10000을 했더니 과적합이 너무 심하게 나서 1000로 했습니다.

File: stock_price_forecast/trainer.py (hunk 4 above)
Comment (byb-8) [reply stock_price_forecast/trainer.py:56]: 정확도 63%정도 나왔습니다.

File: stock_price_forecast/main.py (hunk 3 above)
Comment (kdyeon0309) [reply stock_price_forecast/main.py:14]: 1000으로하니 어떤가요

Comment (byb-8) [reply stock_price_forecast/main.py:14]: 아직 과적합이 나오지만 그 전보다 validation loss 값은 낮습니다.

Comment (kdyeon0309) [reply stock_price_forecast/main.py:14]: epoch 값은 10~100으로 설정하는 것을 개인적으로 추천 드립니다.
그리고 epoch값을 정확하게 지정하는 것이 아닌 100까지의 범위 중 loss값의 변화가 없거나 튀게 되면 자동으로 멈추게 설정하는 방법도 고려해 보는걸 추천해드립니다

Comment (byb-8) [reply stock_price_forecast/main.py:14]: 넵 참고하겠습니다

**Review by kdyeon0309 — APPROVED** [review]
